src/nimony/sem.nim

```diff
@@ -1723,12 +1723,13 @@ type
 
 proc semEnumField(c: var SemContext; n: var Cursor; state: var EnumTypeState)
 
-proc semEnumType(c: var SemContext; n: var Cursor; enumType: SymId; beforeExportMarker: int) =
+proc semEnumType(c: var SemContext; n: var Cursor; enumType: SymId; beforeExportMarker: int): bool =
   # XXX Propagate hasHole somehow
   takeToken c, n
   let magicToken = c.dest[beforeExportMarker]
   var state = EnumTypeState(enumType: enumType, thisValue: createXint(0'i64), hasHole: false,
     isBoolType: magicToken.kind == ParLe and pool.tags[magicToken.tagId] == $BoolT)
+  result = not state.isBoolType
```

Comment:
there is nothing wrong with doing the same for `bool`...

Reply:
I ran into weird problems with `bool` with `declared` last time, the file name didn't seem to exist. Probably it gets fixed by introducing `system.nim`